src/combinatorics/young_diagram.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2022 Jake Levinson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jake Levinson
+-/
+import order.upper_lower
+
+/-!
+# Young diagrams
+
+A Young diagram is a finite set of up-left justified boxes:
+
+[5, 3, 3, 1]
+□□□□□
+□□□
+□□□
+□
```

Comment:
I'm not sure the boxes will render correctly unless they're in a code block
````suggestion
```text
□□□□□
□□□
□□□
□
```
This Young diagram corresponds to the [5, 3, 3, 1] partition of 12.
````